services/motion/builtin/move_attempt.go

```diff
@@ -39,32 +43,38 @@ func newMoveAttempt(ctx context.Context, request *moveRequest) *moveAttempt {
 
 		request:      request,
 		responseChan: make(chan moveResponse),
+
+		position: newReplanner(time.Duration(1000/request.config.PositionPollingFreqHz)*time.Millisecond, request.deviatedFromPlan),
+		obstacle: newReplanner(time.Duration(1000/request.config.ObstaclePollingFreqHz)*time.Millisecond, request.obstaclesIntersectPlan),
 	}
 }
 
 // start begins a new moveAttempt by using its moveRequest to create a plan, spawn relevant replanners, and finally execute the motion.
 // the caller of this function should monitor the moveAttempt's responseChan as well as the replanners' responseChan to get insight
 // into the status of the moveAttempt.
 func (ma *moveAttempt) start() error {
-	plan, err := ma.request.plan(ma.ctx)
+	waypoints, err := ma.request.plan(ma.ctx)
 	if err != nil {
 		return err
 	}
 
+	var waypointIndex atomic.Int32
+	waypointIndex.Store(1)
```

Comment:
If this is intended to be the next target waypoint fi recommend calling it something like `nextTargetWaypoint`

Reply:
Also, shouldn't the `waypointIndex` be a struct member of the moveRequest given that the moveAttempt already has access to the parent moveRequest (and would, by extension have access to this index)?

Reply:
How would you feel about making the `waypoints` & whatever we call the waypoint index struct members on the attemp? Reason being: if you want to write tests that check how far an attempt got during execution & what plan it was executing, currently you don't have any way to determine that other than adding logging in the polling & execute functions.

Reply:
I'm actually not sure if its correct to test the waypoint we were at or the waypoint we are going to.  @biotinker what did you have in mind for this?

Reply:
I still have to write the tests, which will likely introduce this change

Reply:
ErrorState expects this to be the waypoint currently executing, so what we are going to.

Reply:
This has been addressed in my latest changes.  Going to resolve